eppo_core/src/events/event_dispatcher.rs

```diff
@@ -0,0 +1,186 @@
+                                } // else loop to get more events
+                            },
+                            Some(EventDispatcherCommand::Flush) => {
+                                break;
+                            }
+                            Some(EventDispatcherCommand::Exit) => {
+                                // Exit the main loop.
+                                return;
+                            }
+                        }
+                    }
+                }
+            }
+
+            // Send `batch` events.
+            tokio::spawn(async move {
+                // Spawning a new task, so the main task can continue batching events and respond to
+                // commands.
+                if !batch_queue.is_empty() {
+                    batch_queue.make_contiguous();
+                    let (events_to_deliver, _) = batch_queue.as_slices();
+                    EventDispatcher::deliver(&ingestion_url, &events_to_deliver).await;
+                }
+            });
+        }
+    }
+
+    async fn deliver(ingestion_url: &str, events: &[Event]) {
+        // Simulated HTTP request or delivery logic
+        info!(
+            "Pretending to deliver {} events to {}",
+            events.len(),
+            ingestion_url
+        );
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use serde::Serialize;
+    use tokio::sync::mpsc::unbounded_channel;
+    use super::*;
+    use crate::timestamp::now;
+    use tokio::time::Duration;
+    use uuid::Uuid;
+
+    #[derive(Debug, Clone, Serialize)]
+    struct LoginPayload {
+        pub user_id: String,
+        pub session_id: String,
+    }
+
+    #[tokio::test]
+    async fn test_dispatch_starts_delivery() {
+        env_logger::init();
+        let config = EventDispatcherConfig {
+            sdk_key: "test-sdk-key".to_string(),
+            ingestion_url: "http://example.com".to_string(),
+            delivery_interval: Duration::from_millis(100),
+            retry_interval: Duration::from_millis(1000),
+            max_retry_delay: Duration::from_millis(5000),
+            max_retries: Some(3),
+        };
+
+        let (tx, rx) = unbounded_channel();
+        let dispatcher = EventDispatcher::new(config, &tx, 1);
+
+        // Add an event
+        let payload = LoginPayload {
+            user_id: "user123".to_string(),
+            session_id: "session456".to_string(),
+        };
+        let serialized_payload = serde_json::to_value(payload).expect("Serialization failed");
+        dispatcher.dispatch(Event {
+            uuid: Uuid::new_v4(),
+            timestamp: now(),
+            event_type: "test".to_string(),
+            payload: serialized_payload,
+        });
+        tx.send(EventDispatcherCommand::Flush).expect("send should not fail");
+        tx.send(EventDispatcherCommand::Exit).expect("send should not fail");
+
+        dispatcher.event_dispatcher(rx).await;
+        tokio::time::sleep(Duration::from_millis(100)).await;
```

Comment:
confirmed via logs that this works as intended
```
running 1 test
[2025-01-06T22:33:43Z INFO  eppo_core::events::event_dispatcher] Pretending to deliver 1 events to http://example.com
test events::event_dispatcher::tests::test_dispatch_starts_delivery ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 34 filtered out; finished in 0.10s
```

next PR we can hook in an event delivery implementation and assert on that